raiden/smart_contracts/NettingChannelLibrary.sol

```diff
@@ -633,6 +616,20 @@ library NettingChannelLibrary {
         }
     }
 
+    function index_or_throw(Data storage self, address caller_address) private returns (uint8 n) {
```

Comment:
I think that the point of issue #437 is not to abstract away the logic into an `index_or_throw()` function. Instead I believe the point is to use a mapping, from participant address to index, which would be initialized in the constructor of the Netting channel and then can be queried [here](https://github.com/raiden-network/raiden/pull/480/files#diff-09cebda618db5fb2268a55de8de94279R318).

So the implementation of `index_or_throw()` should be a simple mapping lookup, that will throw if the lookup fails.